packages/ifpack2/test/unit_tests/Ifpack2_UnitTestOverlappingRowMatrix.cpp

```diff
@@ -234,15 +234,16 @@ void reducedMatvec(const OverlappedMatrixClass & A,
   auto Y_lcl = Y.getLocalViewDevice ();
   
   // Do the "Local part"
-  auto numLocalRows = undA_lcl.numRows();
+  auto numLocalRows = undA->getNodeNumRows();
   localReducedMatvec(undA_lcl,X_lcl,numLocalRows,Y_lcl);
+
   
   // Now, do the "overlapped part"
   if(overlapLevel > 0) {
     int yrange = hstarts[overlapLevel];
     auto Y_ext = Kokkos::subview(Y_lcl,std::make_pair(numLocalRows,numLocalRows+yrange),Kokkos::ALL());
     
-    int xlimit = numLocalRows + ( (overlapLevel == hstarts.size()-1) ? X_lcl.extent(0) : hstarts[overlapLevel+1] );
+    int xlimit = ( (overlapLevel == hstarts.size()-1) ? X_lcl.extent(0) : numLocalRows+hstarts[overlapLevel+1] );
```

Comment:
Hm, this might not even compile, or it might warn.  `numLocalRows` has type `size_t`, and `size_t` + `int` is `size_t`.  `X_lcl.extent(0)` might be `unsigned int` (for `CudaSpace`) or `size_t` (for `HostSpace`).  The two branches thus may have different types.  Also, both are unsigned, but you're assigning them to signed.